### PR TITLE
fix(input-select): add support for autofill in Chrome

### PR DESCRIPTION
**Pull request by jaredhill4**
https://lambdacurry.atlassian.net/browse/LCL-70

https://user-images.githubusercontent.com/7476478/149038333-af9c9903-c8fc-414d-8dc1-0d8b6f87bee8.mov

### CODE REVIEW

File: packages/ui-components/src/lib/inputs/InputSelect/InputSelect.tsx [hunk 1]
```diff
@@ -120,19 +155,29 @@ export const InputSelect: FC<InputSelectProps> = ({
     setValue(controlledValue);
   }, [controlledValue]);
 
+  const onAnimationStart = ({ animationName }: AnimationEvent) => {
+    if (animationName === 'mui-auto-fill') return setAutoFilling(true);
+    if (animationName === 'mui-auto-fill-cancel') return setAutoFilling(false);
+  };
+
+  // Note: We need to manually handle the browser autofill event by listening for 'animationstart' and handling it based on the
+  // animation name. This is because there is currently no native way to listen explicitly for autofill events.
```
Comment (jaruesink) [inline packages/ui-components/src/lib/inputs/InputSelect/InputSelect.tsx:164]: @jaredhill4 are there helpful documentation or stack overflows for this. Kind mind blowing.

Comment (jaredhill4) [reply packages/ui-components/src/lib/inputs/InputSelect/InputSelect.tsx:164]: @jaruesink,

Yea, there's a few articles:
* https://codedaily.io/tutorials/Animated-Input-Label-with-Chrome-Autofill-Detection-in-React
* https://medium.com/@brunn/detecting-autofilled-fields-in-javascript-aed598d25da7

Comment (jaredhill4) [reply packages/ui-components/src/lib/inputs/InputSelect/InputSelect.tsx]: @jaruesink,

I've made a few more updates and cleaned things up a bit.

**Review by dwene — DISMISSED** [review]
Wow, pretty wild the autofill browser API is so archaic. Nice work figuring it out though.

**Pull request merged**